src/site/content/en/blog/mediastreamtrack-insertable-media-processing/index.md

```diff
@@ -0,0 +1,257 @@
+| 3. Gather feedback & iterate on design   | [In progress](#feedback) |
+| 4. **Origin trial**                      | **[In progress][ot]**    |
+| 5. Launch                                | Not started              |
+
+</div>
+
+## How to use insertable streams for `MediaStreamTrack`
+
+### Enabling via chrome://flags
+
+To experiment with insertable streams for `MediaStreamTrack` locally, without an origin trial token,
+enable the `#enable-experimental-web-platform-features` flag in `chrome://flags`.
+
+### Feature detection
+
+You can feature-detect insertable streams for `MediaStreamTrack` support as follows.
+
+```js
+if ('MediaStreamTrackProcessor' in window && 'MediaStreamTrackGenerator' in window) {
+  // Insertable streams for `MediaStreamTrack` is supported.
+}
+```
+
+### Core concepts
+
+Insertable streams for `MediaStreamTrack` builds on concepts previously proposed by
+[WebCodecs](https://web.dev/webcodecs/) and conceptually splits the `MediaStreamTrack` into two
+components:
+
+- The `MediaStreamTrackProcessor`, which consumes a `MediaStreamTrack` object's source and generates a stream
+  of media frames, specifically [`VideoFrame`](https://w3c.github.io/webcodecs/#videoframe-interface) or
+  [`AudioFrame`](https://w3c.github.io/webcodecs/#audioframe-interface)) objects. You can think of this as a
+  track sink that is capable of exposing the unencoded frames from the track as a `ReadableStream`
+  and that exposes a control channel for signals going in the opposite direction.
+- The `MediaStreamTrackGenerator`, which consumes a stream of media frames and exposes a `MediaStreamTrack`
+  interface, so that it can be used in any sink a `MediaStreamTrack` is currently attached to. It takes
+  video frames as input, and emits control signals that result from subsequent processing.
+
+#### The `MediaStreamTrackProcessor`
+
+A `MediaStreamTrackProcessor` object exposes two properties:
+
+- `readable`: Allows reading the frames flowing through the `MediaStreamTrack`. If the track is a
+  video track, chunks read from `readable` will be `VideoFrame` objects. If the track is an audio
+  track, chunks read from `readable` will be `AudioFrame` objects.
+- `writableControl`: Allows sending control signals to the track. Control signals are objects of
+  type `MediaStreamTrackSignal`.
+
+#### The `MediaStreamTrackGenerator`
+
+A `MediaStreamTrackGenerator` object likewise exposes two properties:
+
+- `writable`: A `WritableStream` that allows writing media frames to the
+  `MediaStreamTrackGenerator`, which is itself a `MediaStreamTrack`. If the `kind` attribute is
+  `"audio"`, the stream accepts `AudioFrame` objects and fails with any other type. If kind is
+  `"video"`, the stream accepts `VideoFrame` objects and fails with any other type. When a frame
+  is written to `writable`, the frame's `close()` method is automatically invoked, so that its
+  internal resources are no longer accessible from JavaScript.
```

Comment:
```suggestion
  is written to `writable`, the frame's `close()` method is automatically invoked, so that its
  media resources are no longer accessible from JavaScript.
```